prediction_market_agent/agents/metaculus_agent/deploy.py

```diff
@@ -1,84 +1,82 @@
             tournament_id=self.tournament_id,
             filter_by=FilterBy.OPEN,
             sort_by=SortBy.NEWEST,
         )
-        logger.info(f"Found {len(markets)} open markets to submit predictions for.")
+        return markets
 
-        if not self.repeat_predictions:
-            # Filter out markets that we have already answered
-            markets = [market for market in markets if not market.have_predicted]
-            logger.info(
-                f"Found {len(markets)} unanswered markets to submit predictions for."
-            )
+    def verify_market(self, market_type: MarketType, market: AgentMarket) -> bool:
+        assert isinstance(
+            market, MetaculusAgentMarket
+        ), "Just making mypy happy. It's true thanks to the check in the `run` method via `supported_markets`."
 
-        for market in markets:
-            logger.info(f"Answering market {market.id}, question: {market.question}")
-            if not self.dummy_prediction:
-                # TODO incorporate 'Resolution criteria', 'Fine print', and
-                # 'Background info' into the prompt given to the agent.
-                answer = agent.answer_binary_market(
-                    market.question, created_time=market.created_time
-                )
-            else:
-                answer = ProbabilisticAnswer(
-                    p_yes=Probability(0.5),
-                    reasoning="Just a test.",
-                    confidence=0.5,
-                )
+        # Filter out the market if the agent isn't configured to re-bet.
+        if not self.repeat_predictions and market.have_predicted:
+            return False
 
-            if answer is None:
-                logger.error("No answer was given. Skipping")
-            else:
-                market.submit_prediction(
-                    p_yes=answer.p_yes,
-                    reasoning=check_not_none(answer.reasoning),
-                )
+        # Otherwise all markets on Metaculus are fine.
+        return True
+
+    def answer_binary_market(self, market: AgentMarket) -> ProbabilisticAnswer | None:
+        assert isinstance(
+            market, MetaculusAgentMarket
+        ), "Just making mypy happy. It's true thanks to the check in the `run` method via `supported_markets`."
+        logger.info(f"Answering market {market.id}, question: {market.question}")
+        answer: ProbabilisticAnswer | None
+        if not self.dummy_prediction:
+            full_question = f"""Question: {market.question}
+Question's description: {market.description}
+Question's fine print: {market.fine_print} 
+Question's resolution criteria: {market.resolution_criteria}"""
+            answer = self.agent.agent.predict(full_question).outcome_prediction
+        else:
+            answer = ProbabilisticAnswer(
+                p_yes=Probability(0.5),
+                reasoning="Just a test.",
+                confidence=0.5,
+            )
```

Comment:
why not `return None`?

Reply:
I don't know, that's your code! 😄 https://github.com/gnosis/prediction-market-agent/pull/310/files#diff-220f7f6f7c82acc320602051ae0cb0a37d96390d7813ce54fde045724c2d3e59R68

Reply:
oh haha! it must be right then...